### PR TITLE
improve documentation

### PR DESCRIPTION
**Pull request by ailox**
Contains no changes in business logic, just documentation and slight refactoring.

* remove unused code blocks
* update copyright years in license header
* describe the purpose of methods
* describe required untyped parameters for functions

Please help me proof-read this before merging.

I have the most doubts about [SignedKeyRegistration and KeyRegistration](https://github.com/ubirch/ubirch-go-udp-client/pull/5/files#diff-8eef5c92219354d7d6ab9fdca08416e1R47), could you check if the comment is correct or can be improved?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,26 +132,25 @@ The possible values are:
...
 
     $ python ./simulation/simulate_http.py "simulation/my-test-input-file.txt"
     
-The input file should contain the data for http requests to the client in hex format. Each line starts with the UUID 
-of the sender followed by any other data (see [Message format](#message-format)).
-
-### Issues
-
-- The configuration from console.demo.ubirch.com sets the msgpack endpoint for 
-  key registration. **Remove the `/mpack` from the end of the keyService URL!**
```
Comment (ailox) [inline README.md:109]: moved this up to configuration, so it is more visible

File: main/client.go [hunk 2]
```diff
@@ -43,6 +44,8 @@ type KeyRegistration struct {
 	ValidNotBefore string `json:"validNotBefore"`
 }
 
+// SignedKeyRegistration is a signed certificate registered with the
+// key-service.
 type SignedKeyRegistration struct {
```
Comment (ailox) [inline main/client.go:49]: I don't know if i understood this correctly; KeyRegistration is when a public key is registered with the key-service, and SignedKeyRegistration is the same, but signed (by whom?)?

File: README.md [hunk 3]
```diff
@@ -1,51 +1,82 @@
-# ubirch-go-udp-client
+# ubirch-client
 
+The ubirch client allows you to quickly integrate the ubirch platform,
+including the anchoring of information into public blockchains, into
+your project.
+
+It provides HTTP and UDP endpoints for sending either hashes that will be
```
Comment (leroxyl) [inline README.md:7]: not implemented yet. currently all messages will be hashed before signing

File: README.md [hunk 4]
```diff
@@ -1,51 +1,82 @@
...
+anchored, or information to be hashed and subsequently anchored.
+
+## UDP Endpoint
+
+
+## HTTP Endpoint
```
Comment (leroxyl) [inline README.md:13]: mixed up http and udp here

File: README.md [hunk 5]
```diff
@@ -1,51 +1,82 @@
...
 ### Configuration
+**NOTE**: The configuration from console.demo.ubirch.com sets the msgpack endpoint for 
+  key registration. **Remove the `/mpack` from the end of the keyService URL!**
+
+#### Via file
+If the `UBIRCH_AUTH` env variable is unset or empty, and the `config.json`
+file exists in the current working dir, the configuration will be read from
+it.
+
+#### Via env
```
Comment (leroxyl) [inline README.md:70]: maybe explain env before file since file config is only second choice

File: README.md [hunk 6]
```diff
@@ -54,21 +85,36 @@ contains the URL to the niomon service (eg. "https://niomon.demo.ubirch.com")
 (eg. "localhost:15002")
```
Comment (leroxyl) [inline README.md:85]: the udp endpoint for signing messages

File: README.md [hunk 7]
```diff
@@ -54,21 +85,36 @@ contains the URL to the niomon service (eg. "https://niomon.demo.ubirch.com")
...
 ###### `UBIRCH_INTERFACE_TXVERIFY`
 (eg. "localhost:15000")
```
Comment (leroxyl) [inline README.md:87]: the udp endpoint for message verification

File: README.md [hunk 8]
```diff
@@ -54,21 +85,36 @@ contains the URL to the niomon service (eg. "https://niomon.demo.ubirch.com")
...
+###### `UBIRCH_AUTH_MAP`
+contains the auth map, which is a mapping from a user UUID to both a key and a auth token.
+The format of this string is:
+
+```json
+{"9080559e-6f62-4122-884a-7a2471a3f635": ["rT7IGJ4zvvvGVN40bIAmGDTyKYq/8gCXqs+vBf3rTLc=", "0f4b2a18-c18d-4bf1-b723-729793521726"]}
+```
+
+* the first uuid is the client UUID registered with the [ubirch-console](https://console.demo.ubirch.com/).
+* the base64 encoded string is a ECDSA secp256k1 key used for signing the UPPs.
```
Comment (leroxyl) [inline README.md:97]: ECDSA prime256v1 key

File: README.md [hunk 9]
```diff
@@ -54,21 +85,36 @@ contains the URL to the niomon service (eg. "https://niomon.demo.ubirch.com")
...
+* the last uuid is the auth-token for the client, as generated by the ubirch console.
```
Comment (leroxyl) [inline README.md:98]: not a UUID (just looks like it). say "the last string"

File: README.md [hunk 10]
```diff
@@ -87,26 +133,25 @@ The possible values are:
 	
 The successfully verified message of the example would then look like this:
 ```
-# UUID                          # timestamp     # some value
 4f6b64a7a5c9483786c00d32bc8e03c080d1a6763192d01500003fc93178a59b00
+|<           UUID             >||< timestamp  >||<      misc    >|
+|          (16 byte)           ||   (8 byte)   ||     (9 byte)   |
```
Comment (leroxyl) [inline README.md:138]: the verifier appends a byte-long code to the original message. if you say the original message is 16+8+X bytes long, the verifier responds with a message thats 16+8+X+2 bytes long (not 9 byte)

**Comment by ailox** [comment]
Stale branch

**Pull request closed without merging**